components/organisation-unit-tree/src/organisation-unit-node/use-org-data/use-org-data.js

```diff
@@ -35,35 +35,25 @@ export const useOrgData = (
 
     const defaultData = { id, displayName }
     const variables = { id, isUserDataViewFallback }
-    const [state, setState] = useState({
-        loading: true,
-        error: null,
-        data: defaultData,
+    const { loading, error, data } = useDataQuery(ORG_DATA_QUERY, {
+        variables
     })
 
-    useDataQuery(ORG_DATA_QUERY, {
-        variables,
-        onComplete: ({ orgUnit: node }) => {
-            const sorted = !suppressAlphabeticalSorting
-                ? sortNodeChildrenAlphabetically(node)
-                : node
+    const transformedData = useMemo(() => {
+        if (!data) {
+            return defaultData
+        }
 
-            const nextData = { ...defaultData, ...sorted }
-            setState({
-                ...state,
-                data: nextData,
-                loading: false,
-            })
-            onComplete && onComplete(nextData)
-        },
-        onError: queryError => {
-            setState({
-                ...state,
-                error: queryError,
-                loading: false,
-            })
-        },
-    })
+        const { orgUnit: node } = data
+        const transformed = suppressAlphabeticalSorting ? node : sortNodeChildrenAlphabetically(node)
+        const merged = { ...defaultData, ...transformed }
+
+        return merged
+    }, [data, suppressAlphabeticalSorting, id, displayName])
+
+    useEffect(() => {
+        onComplete && onComplete(transformedData)
+    }, [onComplete, transformedData])
```

Comment:
I'm using a useEffect here instead of including it in the useMemo as it seems more suited to a side-effect like this, and not _directly_ related to the memoized value. Seems better to leave this to react's scheduling.